collector/src/lib.rs

```diff
@@ -703,19 +693,25 @@ mod tests {
         random()
     }
 
+    fn collection_uri_regex_matcher(task_id: &TaskId) -> Matcher {
```

Comment:
It appears this function is only used in tests -- IMO move it into the `tests` module or otherwise mark it `#[cfg(test)]`.

Reply:
It's already in `mod tests` and so should not be defined outside `#[cfg(test)]`.